### DISCUSSION TITLE
API V3 definitions of input parameters yet?

### DISCUSSION THREAD

**Discussion by hubert-associates** [NREL/REopt-API-Analysis#83] [8 - API: Application Programming Interface]
Hi. I know that API V3 is still work in progress. Just wondering if I missed a place where the inputs are defined other than the .jl code? I see https://nrel.github.io/REopt.jl/dev/reopt/inputs/ . Code viewing helps in some cases, but not all. https://nrel.github.io/REopt.jl/dev/reopt/outputs/ is more elaborated and helps.

For example. What does "tou_" mean, and what are "blended_energy_rates / blended_demand_rates" ?

Thanks for any pointers.

**Comment by LParkhill** [comment]
@hubert-associates Thanks for this very useful observation. We will prioritize getting the input definitions listed in a clear and easily located fashion similar to the output definitions. 

In the meantime, you can hopefully find most of the definitions at the older API site: https://github.com/NREL/REopt_API/blob/master/reo/nested_inputs.py 
(tou refers to utility rates with specific time-of-use variation - usually by day and by hour, while blended rates use a single energy or demand rate for the entire year or for an entire month)

**Reply by LParkhill** [reply]
@hubert-associates the developers had another temporary solution for input descriptions: 

- "Also, they can do a GET request to this URL to get a detailed list with descriptions for all the inputs: https://developer.nrel.gov/api/reopt/dev/help/?api_key= + API_KEY."

This should give a more complete and relevant list than the previous suggestion.
